gensim/models/phrases.py

```diff
@@ -311,6 +353,20 @@ def __getitem__(self, sentence):
 
         return [utils.to_unicode(w) for w in new_s]
 
+    # calculation of score based on original mikolov word2vec paper
+    # len_vocab and min_count set so functools.partial works
+    @staticmethod
+    def original_scorer(worda_count, wordb_count, bigram_count, len_vocab=0.0, min_count=0.0):
+        return (bigram_count - min_count) / worda_count / wordb_count * len_vocab
```

Comment:
Beware of integer divisions - this code is brittle.

Reply:
I didn't fix this in PR #1573 . Rather, I just cast everything before calling the scoring method in Phrases and Phraser. I think that's the better place to do the casting since then it fixes the problem for all custom scorers as well. 

Of course, I can do the casting in the scoring methods as well. Let me know if you still think I need it here and in `npmi_scorer` and I'll update PR #1573. It's extra steps, but I'd assume the performance hit is infinitesimal.